### PR TITLE
Remove deprecated api settings

### PR DESCRIPTION
**Pull request by mzitnik**
## Summary
* remove deprecated API load, dump, and connect
* remove use_no_proxy settings

## Checklist
Delete items not relevant to your PR:
- [X] Unit and integration tests covering the common scenarios were added
- [X] A human-readable description of the changes was provided to include in CHANGELOG
- [ ] For significant changes, documentation in https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

**Comment by mzitnik** [comment]
@zhicwu if you can take a look i have made some cleanup

**Comment by zhicwu** [comment]
There are many changes. Will review, test, and merge tonight.

File: clickhouse-grpc-client/src/main/java/com/clickhouse/client/grpc/ClickHouseGrpcChannelFactory.java [hunk 1]
```diff
@@ -193,9 +193,6 @@ protected void setupMisc() {
             builder.enableFullStreamDecompression();
         }
 
-        if (config.isUseNoProxy()) {
```
Comment (zhicwu) [inline clickhouse-grpc-client/src/main/java/com/clickhouse/client/grpc/ClickHouseGrpcChannelFactory.java:196]: Let's simply replace `config.isUseNoProxy()` to `config.getProxyType() == ClickHouseProxyType.DIRECT`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -179,7 +179,7 @@ jobs:
       matrix:
         # most recent LTS releases as well as latest stable builds
         # https://github.com/ClickHouse/ClickHouse/pulls?q=is%3Aopen+is%3Apr+label%3Arelease
-        clickhouse: ["22.3", "22.8", "23.3", "latest"]
+        clickhouse: ["22.8", "23.3", "23.7", "latest"]
```
Comment (zhicwu) [inline .github/workflows/build.yml:182]: If we wait for a few more days, we can change 23.7 to 23.8 lts :)

**Review by zhicwu — CHANGES_REQUESTED** [review]

**Comment by mzitnik** [comment]
@zhicwu do you know why SonarCloud Quality Gate failed? (nothing was changed just removed deprecated api)

**Review by zhicwu — APPROVED** [review]

**Comment by zhicwu** [comment]
> @zhicwu do you know why SonarCloud Quality Gate failed? (nothing was changed just removed deprecated api)

The coverage of changed lines is less than 80%. It's the default rule but not mandatory.

**Pull request merged**